framework/src/tinyalsa/tinyalsa.c

```diff
@@ -476,7 +475,7 @@ int pcm_writei(struct pcm *pcm, const void *data, unsigned int frame_count)
 		return -EINVAL;
 	}
 
-	if (pcm->flags & PCM_IN) {
+	if (!(pcm->flags & PCM_OUT)) {
```

Comment:
There's no reason to change this.

Reply:
The description of the API says "This function is only valid for PCMs opened with the @ref PCM_OUT flag". So I think it is better to change like this.